### PR TITLE
Flask: add GET /api/enums

### LINKED ISSUE #39: GET /api/enums
Example uses:
- GET /api/enums?type=DatasetType,DatasetCondition
- GET /api/enums?fortable=Dataset,Participant

Input:
- Optional array of which enums to return

Output:
- Object: fields for each requested enum (or all) that are arrays of accepted values

### PR DESCRIPTION
**Pull request by Madelinehazel**
closes #39

### CODE REVIEW

**Review by kevinlul — CHANGES_REQUESTED** [review]
I don't think this really fulfils the requirements of /api/enums because you have to know the name beforehand and fetch each one individually.

**Comment by Madelinehazel** [comment]
returns: 
{
  "AnalysisState": [
    "Requested", 
    "Running", 
    "Done", 
    "Error", 
    "Cancelled"
  ], 
  "DatasetCondition": [
    "Control", 
    "GermLine", 
    "Somatic"
  ], 
  "DatasetExtractionProtocol": [
    "Something"
  ], 
  "DatasetReadType": [
    "PairedEnd", 
    "SingleEnd"
  ], 
  "DatasetType": [
    "CES", 
    "CGS", 
    "CPS", 
    "RES", 
    "RGS", 
    "RLM", 
    "RMM", 
    "RRS", 
    "RTA", 
    "WES", 
    "WGS", 
    "RNASeq", 
    "RCS", 
    "RDC", 
    "RDE"
  ], 
  "ParticipantType": [
    "Proband", 
    "Mother", 
    "Father", 
    "Sibling"
  ], 
  "Sex": [
    "Male", 
    "Female", 
    "Other"
  ], 
  "TissueProcessing": [
    "FF", 
    "FFPE"
  ], 
  "TissueSampleType": [
    "Blood", 
    "Saliva", 
    "Lymphocyte", 
    "Fibroblast", 
    "Muscle", 
    "Skin", 
    "Urine", 
    "Plasma", 
    "Unknown", 
    "Kidney"
  ]
}

File: flask/app/routes.py [hunk 1]
```diff
@@ -329,3 +330,13 @@ def datasets_list():
...
     return jsonify(datasets)
+
+
+@app.route('/api/enums', methods = ['GET'])
+@login_required
+def get_enums():
+    enums = {}
+    for name, obj in inspect.getmembers(models, inspect.isclass):
+        if issubclass(obj, Enum) and name != 'Enum':
+                enums[name] = [e.value for e in getattr(models, name)]
```
Comment (kevinlul) [inline flask/app/routes.py:341]: Looks like there's a double indent here

**Review by kevinlul — CHANGES_REQUESTED** [review]
This is pretty neat!

**Review by kevinlul — APPROVED** [review]
LGTM

**Review by delvinso — APPROVED** [review]

**Pull request merged**